plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtRestoreBackupCommandWrapper.java

```diff
@@ -0,0 +1,181 @@
+                               String diskType, Long deviceId, String volumeUUID) {
+        String mountDirectory = mountBackupDirectory(backupRepoAddress, backupRepoType);
+        Pair<String, String> bkpPathAndVolUuid;
+        try {
+            bkpPathAndVolUuid = getBackupPath(mountDirectory, volumePath, backupPath, diskType, deviceId.intValue(), volumeUUID);
+            try {
+                replaceVolumeWithBackup(volumePath, bkpPathAndVolUuid.first());
+            } catch (IOException e) {
+                throw new CloudRuntimeException(String.format("Unable to revert backup for volume [%s] due to [%s].", bkpPathAndVolUuid.second(), e.getMessage()), e);
+            }
+        } catch (Exception e) {
+            throw new CloudRuntimeException("Failed to restore volume", e);
+        } finally {
+            unmountBackupDirectory(mountDirectory);
+            deleteTemporaryDirectory(mountDirectory);
+        }
+    }
+
+
+    private String mountBackupDirectory(String backupRepoAddress, String backupRepoType) {
+        String randomChars = RandomStringUtils.random(5, true, false);
+        String mountDirectory = String.format("%s.%s",BACKUP_TEMP_FILE_PREFIx , randomChars);
+        try {
+            mountDirectory = Files.createTempDirectory(mountDirectory).toString();
+            String mount = String.format(MOUNT_COMMAND, backupRepoType, backupRepoAddress, mountDirectory);
+            Script.runSimpleBashScript(mount);
+        } catch (Exception e) {
+            throw new CloudRuntimeException(String.format("Failed to mount %s to %s", backupRepoType, backupRepoAddress), e);
+        }
+        return mountDirectory;
+    }
+
+    private void unmountBackupDirectory(String backupDirectory) {
+        try {
+            String umountCmd = String.format(UMOUNT_COMMAND, backupDirectory);
+            Script.runSimpleBashScript(umountCmd);
+        } catch (Exception e) {
+            throw new CloudRuntimeException(String.format("Failed to unmount backup directory: %s", backupDirectory), e);
+        }
+    }
+
+    private void deleteTemporaryDirectory(String backupDirectory) {
+        try {
+            Files.deleteIfExists(Paths.get(backupDirectory));
+        } catch (IOException e) {
+            throw new CloudRuntimeException(String.format("Failed to delete backup directory: %s", backupDirectory), e);
+        }
+    }
+
+    private Pair<String, String> getBackupPath(String mountDirectory, String volumePath, String backupPath, String diskType, int deviceId, String volumeUuid) {
+        String bkpPath = String.format(FILE_PATH_PLACEHOLDER, mountDirectory, backupPath);
+        int lastIndex = volumePath.lastIndexOf("/");
+        String volUuid = Objects.isNull(volumeUuid) ? volumePath.substring(lastIndex + 1) : volumeUuid;
+        String backupFileName = String.format("%s.%s.%s.qcow2", deviceId, diskType.toLowerCase(Locale.ROOT), volUuid);
```

Comment:
That's a limitation of the B&R plugin, it will be always qcow2 - see docs at https://github.com/apache/cloudstack-documentation/pull/429 - this can change in future.